pybamm/models/base_model.py

```diff
@@ -118,6 +118,11 @@ def __init__(self, name="Unnamed model"):
         # Default timescale is 1 second
         self.timescale = pybamm.Scalar(1)
         self.length_scales = {}
+        self.timescale_eval = self.timescale.evaluate()
+        self.length_scales_eval = {
+            domain: scale.evaluate()
+            for domain, scale in self.length_scales.items()
+        }
```

Comment:
This is scary. With a model that isn't the base model, you would end up with timescale being some parameter and timescale_eval being equal to 1 (until you update it in the solver) which could get confusing

Reply:
There were some tests that failed without adding this that weren't battery models and didn't have a solver. I agree though it's a bit dodgy and is now solved by checking for the attribute